### PR TITLE
Metadata upload package

### PR DESCRIPTION
**Pull request by kimberlyh66**
First version of metadata upload package for betydb. Package files are in metadata folder.

### CODE REVIEW

**Comment by KristinaRiemer** [comment]
I'm reviewing this.

**Comment by kimberlyh66** [comment]
Okay. Thanks. I just changed the name of the package folder to metadata.

**Comment by KristinaRiemer** [comment]
@kimberlyh66 I'm getting the following error when I try to run the example in the help page for `add_row`. It looks like it's coming from the second line of the function, but I don't know enough about what it's doing to debug it. 

```
Error in inherits(.data, "grouped_df") : 
  argument ".data" is missing, with no default
```

**Comment by KristinaRiemer** [comment]
A couple thoughts about function names. `add_row` is a function name in some other commonly-used R packages. While it's okay to have the same function names because the package can be specified with double colons (i.e., `betydbMetadata::add_row`), it can be less confusing for users if that's not necessary. I would maybe rename that function to `add_row_bety` or something? 

Relatedly, `add_specie` maybe should be `add_species`. This is also a function in another package, so maybe it should actually follow the same format as the add row function and be `add_species_bety`?

File: metadata/R/associate_vals.R [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+#' @param dbcon Database connection, as created by [RPostgres::dbConnect]
+#' @param commit Value indicating whether to commit database transaction;
+#' (boolean: TRUE or FALSE, default is FALSE)
+#'
+#' @return Exits on success
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' associate_vals(tbl_name = 'experiment_sites', col_1 = 'experiment_id',
```
Comment (KristinaRiemer) [inline metadata/R/associate_vals.R:19]: `dbcon = dbcon` argument is missing from this example.

**Comment by kimberlyh66** [comment]
In reference to your comment, I think the problem with the `add_row` example is that `add_row` is also the name of a function from the `tibble` package.  I think the error that you are getting is outputted from the tibble package.

```
Error in inherits(.data, "grouped_df") : 
  argument ".data" is missing, with no default
```

Do you get same error when you run with `betydbMetadata::add_row()`?

I am not sure if you will be able to run any of these examples on the help page since the bety docker is not working correctly. I am getting an error when I run the add_row example.

**Comment by KristinaRiemer** [comment]
Oooh, specifying the package worked. I definitely think `add_row` should be renamed then. It is returning `[1] TRUE` though, I don't know if it's supposed to? 

`associate_vals` is throwing this error. 

```
Error in result_create(conn@ptr, statement) : 
  Failed to prepare query: ERROR:  relation "experiment_sites" does not exist
LINE 1: insert into experiment_sites
```

`add_specie` and `get_fields` seem to be working, or at least aren't returning errors. I brought this up with the metadata upload tutorial, but is there a way to show that these functions actually worked and added the right content to the test database?

**Comment by KristinaRiemer** [comment]
`get_params` and `get_statement` also appear to work.

**Comment by kimberlyh66** [comment]
Yes, I think you are right. I think we should work on improving the names of the functions.

I think you are getting the error for `associate_vals` since the bety docker is not working correctly. I think that something has changed with the schema of the test database. 

You can test if data is uploaded correctly by opening a terminal and running queries on psql command line after the function is run. For example, you could run `select * from experiments where name = 'MAC Season 8: Border'` if you just added the experiment `MAC Season 8: Border`. But, I am not sure if this is the best approach for the tutorial. Let me know if you have any suggestions.

Also, make sure to set the `commit` param to `TRUE` when calling `add_row` or `associate_val`, or else data will not be uploaded.

**Comment by KristinaRiemer** [comment]
Okay, we probably should wait until the Docker issues are fixed, and the functions are working, before merging this. 

We might want to talk with @dlebauer about how to make sure data uploaded correctly, because I see the next big steps as: 1) write tests for these functions to make sure they're doing what we expect and 2) integrate the package with Travis to do that automatically (which will mean putting the stuff to get the postgres Docker up and running into the travis.yml). I'm happy to help with or do any of these things.

**Comment by kimberlyh66** [comment]
Okay. Thanks for your review. I'll probably need help with writing the tests and integrating with Travis.

**Comment by kimberlyh66** [comment]
Closing this pull request - changes need to be made to package.

**Pull request closed without merging**